robot_skills/src/robot_skills/arms.py

```diff
@@ -318,6 +322,188 @@ class GripperState(object):
+        :param timeout: timeout in seconds
+        :return: True or False
+        """
+        return self._exchange_with_human(False, timeout)
+
+    def handover_to_robot(self, timeout=10):
+        # type: (float) -> bool
+        """
+        Handover an item from a human to the robot.
+
+        Feels if user slightly pushes an item in the gripper. On timeout, it will return False.
+        :param timeout: timeout in seconds
+        :return: True or False
+        """
+        return self._exchange_with_human(True, timeout)
+
+    def _exchange_with_human(self, to_robot, timeout=10.0):
+        # type: (bool, float) -> bool
+        """
+        Exchange an item between the arm and a human.
+
+        If 'to_robot', it feels if user slightly pushes an item in the gripper.
+        If not 'to_robot', it feels if user slightly pulls or pushes (the item in) the arm.
+
+        :param to_robot: Direction of the exchange.
+        :param timeout: Timeout in seconds.
+        :return: Whether the exchange succeeded.
+        """
+        if to_robot:
+            topic = '/{}/handoverdetector_{}/toggle_human2robot'.format(self.robot_name, self.side)
+        else:
+            topic = '/{}/handoverdetector_{}/toggle_robot2human'.format(self.robot_name, self.side)
+
+        pub = rospy.Publisher(topic, std_msgs.msg.Bool, queue_size=1, latch=True)
+        pub.publish(std_msgs.msg.Bool(True))
+
+        result_topic = '/{}/handoverdetector_{}/result'.format(self.robot_name, self.side)
+        try:
+            rospy.wait_for_message(result_topic, std_msgs.msg.Bool, timeout)
+            return True
+        except rospy.ROSException as e:
+            rospy.logerr(e)
+            return False
+
+class ArmDatabasePiece(ArmPiece):
+    """
+    High level knowledge about arm properties.
+
+    :var goals: Default poses of the arm.
+    :var trajectories: Default trajectories of the arm.
+    """
+    def __init__(self, robot_name):
+        # type: (str)
+        super(ArmDatabasePiece, self).__init__(robot_name)
+
+        self.goals = self.load_param('skills/arm/default_configurations')
+        self.trajectories = self.load_param('skills/arm/default_trajectories')
+
+    def get_goal(self, configuration):
+        # type: (str)
+        """
+        Get an arm goal by name.
+
+        :param configuration: Name of the configuration to retrieve.
+        :return: Configuration for the arm, or None if the requested configuration does not exist.
+        """
+        return self.goals.get(configuration)
+
+    def get_trajectory(self, configuration):
+        # type: (str)
+        """
+        Get an arm trajectory by name.
+
+        :param configuration: Name of the trajectory to retrieve.
+        :return: Trajectory for the arm, or None if the requested trajectory does not exist.
+        """
+        return self.trajectories.get(configuration)
+
+class ArmMarkerPublisherPiece(ArmPiece):
```

Comment:
Ik noemde al de CartesianInterface in een voorgaande comment. Deze functionaliteit zou daar onderdeel van kunnen zijn (`publish_marker` wordt nu alleen aangeroepen vanuit de `send_goal` functie)